### ISSUE TITLE
Handle relative URLs in Link header

### ISSUE THREAD

**Issue by glb** [simonw/paginate-json#5]
It would be super-handy if this tool were able to handle relative URLs in the link header; something like:

```
Link: </widgets?page=2>;rel="next"
```

which would be joined with the original URL. This would be relatively straightforward to implement using `requests.compat.urljoin`:

```diff
diff --git a/paginate_json/cli.py b/paginate_json/cli.py
index 4be0fb4..555577d 100644
--- a/paginate_json/cli.py
+++ b/paginate_json/cli.py
@@ -63,7 +63,11 @@ def paginate(
                 json.dumps(dict(response.headers), indent=4, default=repr), err=True
             )
         try:
-            url = response.links.get("next").get("url")
+            # Get the 'next' link
+            next = response.links.get("next").get("url")
+
+            # Resolve the potentially-relative URL to be an absolute URL
+            url = requests.compat.urljoin(url, next)
         except AttributeError:
             url = None
         if jq:
```

Thoughts? Would you be open to a PR that implements this?

**Comment by simonw** [comment]
This seems like a good idea to me.

**Comment by simonw** [comment]
Released in 0.3.1: https://github.com/simonw/paginate-json/releases/tag/0.3.1

**Comment by glb** [comment]
Thank you!
